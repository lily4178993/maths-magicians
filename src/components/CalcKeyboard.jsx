import React from 'react';
import PropTypes from 'prop-types';
import CalcButton from './CalcButton';

const CalcKeyboard = ({ buttonProps }) => (
  <div className="keyboard">
    {buttonProps.map((buttonProp) => (
      <CalcButton
        key={buttonProp.id}
        symbol={buttonProp.symbol}
      />
    ))}
  </div>
);

CalcKeyboard.propTypes = {
  buttonProps: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      symbol: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default CalcKeyboard;
