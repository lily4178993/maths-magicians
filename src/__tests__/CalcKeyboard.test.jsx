import React from 'react';
import { render } from '@testing-library/react';
import CalcKeyboard from '../components/CalcKeyboard';

test('PropTypes snapshot', () => {
  const buttonProps = [
    { id: 3, symbol: '1' },
    { id: 3, symbol: '2' },
    { id: 1, symbol: '+' },
    { id: 2, symbol: '-' },
  ];
  const buttonEvent = jest.fn();

  const { asFragment } = render(
    <CalcKeyboard buttonProps={buttonProps} buttonEvent={buttonEvent} />,
  );
  expect(asFragment()).toMatchSnapshot();
});
