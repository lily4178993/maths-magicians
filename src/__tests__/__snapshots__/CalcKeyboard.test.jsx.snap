// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropTypes snapshot 1`] = `
<DocumentFragment>
  <div
    class="keyboard"
  >
    <button
      title="1"
      type="button"
    >
      1
    </button>
    <button
      title="2"
      type="button"
    >
      2
    </button>
    <button
      class="orange"
      title="+"
      type="button"
    >
      +
    </button>
    <button
      class="orange"
      title="-"
      type="button"
    >
      -
    </button>
  </div>
</DocumentFragment>
`;
